homeassistant/components/steam_online/config_flow.py

```diff
@@ -0,0 +1,222 @@
+                {
+                    vol.Required(
+                        CONF_API_KEY, default=user_input.get(CONF_API_KEY) or ""
+                    ): str,
+                    vol.Required(
+                        CONF_ACCOUNT, default=user_input.get(CONF_ACCOUNT) or ""
+                    ): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_config: ConfigType) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        for entry in self._async_current_entries():
+            if entry.data[CONF_API_KEY] == import_config[CONF_API_KEY]:
+                return self.async_abort(reason="already_configured")
+        _msg = "Steam yaml config in now deprecated and has been imported. Please remove it from your config"
+        LOGGER.warning(_msg)
+        import_config[CONF_ACCOUNT] = import_config[CONF_ACCOUNTS][0]
+        return await self.async_step_user(import_config)
+
+    async def async_step_reauth(self, user_input: dict[str, str]) -> FlowResult:
+        """Handle a reauthorization flow request."""
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Confirm reauth dialog."""
+        if user_input is not None:
+            return await self.async_step_user()
+
+        self._set_confirm_only()
+        return self.async_show_form(step_id="reauth_confirm")
+
+
+class SteamOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle Steam client options."""
+
+    def __init__(self, entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.entry = entry
+        self.options = dict(entry.options)
+
+    async def async_step_init(
+        self, user_input: dict[str, dict[str, str]] | None = None
+    ) -> FlowResult:
+        """Manage Steam options."""
+        if user_input is None:
+            try:
+                interface = await self.hass.async_add_executor_job(
+                    steam.api.interface, "ISteamUser"
+                )
+                friends = await self.hass.async_add_executor_job(
+                    ft.partial(
+                        interface.GetFriendList, steamid=self.entry.data[CONF_ACCOUNT]
+                    )
+                )
+                friends = (
+                    await self.hass.async_add_executor_job(friends.get, "friendslist")
+                )["friends"]
+                _users_str = str([user["steamid"] for user in friends])
+                names = await self.hass.async_add_executor_job(
+                    ft.partial(interface.GetPlayerSummaries, steamids=_users_str)
+                )
+                names = (await self.hass.async_add_executor_job(names.get, "response"))[
```

Comment:
Instead of calling into the executor many times, create a single method that does all this logic and pass it to the executor.

Reply:
Use that guideline any time you're doing 2+ calls.

Reply:
Your comments describe my thought process when I was making this but the supporting library still seems to block when we are accessing dictionary values. Maybe I can condense this into two executor calls. My plans for twitch are much cleaner than this.

Reply:
Blocking calls won't matter as long as it is all in the executor.

Reply:
Seems to be working now